docs/explanation/core/proposer-based-timestamps.md

```diff
@@ -55,44 +33,182 @@ The following protocols and application features require a reliable source of ti
 * IBC packets can use either a [timestamp or a height to timeout packet
  delivery](https://ibc.cosmos.network/v8/ibc/light-clients/updates-and-misbehaviour?_highlight=time#checkformisbehaviour).
 
-### Proposer Selects a Block Timestamp
+### Enabling PBTS
+
+The PBTS algorithm is **not enabled by default** in CometBFT v1.0.
+
+If a network upgrades to CometBFT v1.0, it will still use the BFT Time
+algorithm until PBTS is enabled.
+The same applies to new networks that do not change the default values for
+consensus parameters in the genesis file.
+
+Enabling PBTS requires configuring the [consensus parameters](#consensus-parameters)
+that govern the operation of the algorithm.
+
+### Selecting a Timestamp
 
-When the proposer node creates a new block proposal, the node reads the time
-from its local clock and uses this reading as the timestamp for the proposed
-block.
+When a validator creates a new block, it reads the time from its local clock
+and uses this reading as the timestamp for the block.
+The proposer of a block is thus free to select the block timestamp, but this
+timestamp has to be validated by other nodes in the network.
 
-### Timeliness
+### Validating Timestamps
 
-When each validator on the network receives a proposed block, it
-performs a series of checks to ensure that the block can be considered valid as
-a candidate to be the next block in the chain.
+When each validator on the network receives a proposed block, it performs a
+series of checks to ensure that the block can be considered valid as a
+candidate to be the next block in the chain.
+If the block is considered invalid, the validator issues a `nil` prevote,
+signaling to the rest of the network that the proposed block is not valid.
 
 The PBTS algorithm performs a validity check on the timestamp of proposed
 blocks. When a validator receives a proposal it ensures that the timestamp in
-the proposal is within a bound of the validator's local clock. Specifically, the
-algorithm checks that the timestamp is no more than `Precision` greater than the
-node's local clock and no less than `Precision + MessageDelay` behind than the
-node's local clock. This creates range of acceptable timestamps around the
-node's local time. If the timestamp is within this range, the PBTS algorithm
-considers the block **timely**. If a block is not **timely**, the node will
-issue a `nil` prevote for this block, signaling to the rest of the network
-that the node does not consider the block to be valid.
+the proposal is within a bound of the validator's local clock.
+For that it uses `Precision` and `MessageDelay` consensus parameters, 
+ which are the same across all nodes.
+Specifically, the algorithm checks that the timestamp is
+no more than `Precision` greater than the node's local clock
```

Comment:
I think this is backwards, no? Shouldn't it be no less than timestamp - precision, and no more than message delay + precision?

Reply:
The block timestamp cannot be in the future, that is, greater than the current clock time. We add precision because clocks are not perfectly synchronized.

So, receiveTime >= timestamp - precision, or timestamp <= receiveTime + precision..

Is this logic right?